File: tiq-e/strings/5/main.py
from time import perf_counter


class Solution:
    def isPalindrome(self, s):
        s = ''.join(c for c in s if c.isalnum()).lower()
        return s == s[::-1]


s = "ab_a"

start = perf_counter()
result = Solution().isPalindrome(s)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/strings/5/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn is_palindrome(s: String) -> bool {
        let s: String = s
            .to_lowercase()
            .chars()
            .filter(|c| c.is_alphanumeric())
            .collect();
        s.chars()
            .zip(s.chars().rev())
            .fold(true, |acc, (a, b)| acc & (a == b))
    }
}

fn main() {
    let s = "ab_a".to_string();

    let start = Instant::now();
    let result = Solution::is_palindrome(s);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}
